kai/rpc_server/server.py

```diff
@@ -199,9 +199,18 @@ def initialize(
                 app.config.model_provider, app.config.demo_mode, cache
             )
             cache.model_id = re.sub(r"[\.:\\/]", "_", model_provider.model_id)
+
+            model_provider.validate_environment()
         except Exception as e:
-            app.log.error("unable to get model provider:", e)
-            raise
+            server.shutdown_flag = True
```

Comment:
Does shutdown flag mean we are shutting down the server?

Reply:
Yes, though I'm open to having it stay on

Reply:
I would think the end goal would be:
1. User tries to start the server with bad arguments via the IDE
2. Initialize fails, returns an error, but kai keeps running.
3. IDE tells the user what they did wrong.
4. User fixes.
5. User tries to start the server again. IDE recognizes kai is already started. IDE just sends 'initialize' request with updated params.
6. Success.